external/avro/src/main/scala/org/apache/spark/sql/avro/SchemaConverters.scala

```diff
@@ -139,7 +142,16 @@ object SchemaConverters {
 
       case FloatType => builder.floatType()
       case DoubleType => builder.doubleType()
-      case _: DecimalType | StringType => builder.stringType()
+      case StringType => builder.stringType()
+      case d: DecimalType =>
+        val avroType = LogicalTypes.decimal(d.precision, d.scale)
+        if (nullable) {
+          val schema = avroType.addToSchema(SchemaBuilder.builder().bytesType())
+          builder.`type`(schema)
```

Comment:
For decimal, it requires to set three keys in props: `logicalType`, `precision`, `scale`. 
So here I choose to use such way, instead of setting properties like logical Timestamp/Date type.

Reply:
are the precision and scale prop key public? If they are, we should still set the props. It's more important to keep the code consistent.

Reply:
No, the keys are private. Also there is no method like `fixedBuilder` (like `builder.longBuilder().prop(...)`)